magnum_cluster_api/resources.py

```diff
@@ -2048,7 +2048,10 @@ def get_object(self) -> dict:
                                             "template": textwrap.dedent(
                                                 """\
                                                     - cidr: {{ .nodeCidr }}
-                                                      dnsNameservers: {{ .dnsNameservers }}
+                                                      dnsNameservers:
+                                                      {{- range .dnsNameservers }}
+                                                        - {{ . }}
+                                                      {{- end }}
                                                     """
```

Comment:
I think we should create a functional test for this to test the behaviour when 1 or many DNS servers are being included and make sure that it propagates properly.  There are examples of this already